.golangci.yaml.in

```diff
@@ -0,0 +1,204 @@
+          alias: $1$2
+        - pkg: github.com/gardener/gardener-extension-registry-cache/pkg/(\w+)/apis/config/(v\d[\w\d]*)
+          alias: ${1}config${2}
+        - pkg: github.com/gardener/gardener-extension-registry-cache/pkg/(\w+)/apis/config/(v\d[\w\d]*)/([a-z]+)
+          alias: $1$3
+        - pkg: github.com/gardener/gardener-extension-registry-cache/pkg/\w+/controller/([\w\d]+)
+          alias: ${1}controller
+        - pkg: github.com/gardener/gardener-extension-registry-cache/pkg/\w+/webhook/([\w\d]+)
+          alias: ${1}webhook
+        - pkg: github.com/gardener/gardener-extension-registry-cache/pkg/utils/(\w+)
+          alias: ${1}utils
+        - pkg: github.com/gardener/gardener-extension-registry-cache/pkg/admission/cmd
+          alias: admissioncmd
+        - pkg: github.com/gardener/gardener-extension-registry-cache/pkg/admission/validator/(\w+)
+          alias: ${1}validator
+        - pkg: github.com/gardener/gardener-extension-registry-cache/pkg/cmd
+          alias: registrycmd
+    loggercheck:
+      require-string-key: true
+      no-printf-like: true
+    misspell:
+      mode: restricted # only check comments
+    revive:
+      rules:
+        - name: duplicated-imports
+        - name: unused-parameter
+        - name: unreachable-code
+        - name: context-as-argument
+        - name: early-return
+        - name: exported
+    custom:
+      logcheck:
+        path: <<LOGCHECK_PLUGIN_PATH>>/logcheck.so
+        description: Check structured logging calls to logr.Logger instances
+        original-url: github.com/gardener/gardener/hack/tools/logcheck
+  exclusions:
+    generated: lax
+    rules:
+      - linters:
+          - staticcheck
+        text: 'SA1019:' # Excludes messages where deprecated variables are used
+      - linters:
+          - staticcheck
+        text: 'ST1001:' # should not use dot imports
+      # errcheck: Almost all programs ignore errors on these functions and in most cases it's ok
+      - path: (.+)\.go$
+        text: Error return value of .((os\.)?std(out|err)\..*|.*Close|.*Flush|os\.Remove(All)?|.*print(f|ln)?|os\.(Un)?Setenv). is not checked
+      # revive:
+      - path: (.+)\.go$
+        text: var-naming # ((var|const|struct field|func) .* should be .*
+      - path: (.+)\.go$
+        text: dot-imports # should not use dot imports
+      - path: (.+)\.go$
+        text: package-comments # package comment should be of the form
+      - path: (.+)\.go$
+        text: unexported-return # exported func .* returns unexported type .*, which can be annoying to use
+      - path: (.+)\.go$
+        text: indent-error-flow # if block ends with a return statement, so drop this else and outdent its block
+      - path: (.+)\.go$
+        text: 'exported: (type|func) name will be used as .* by other packages, and that stutters;'
+      - path: (.+)\.go$
+        text: exported function Convert_.* should have comment or be unexported
```

Comment:
In the registry-cache, we don't have a non-generated `Convert_*` funcs. The generated ones, all of them have a comment. 
I would personally drop this exclusion. In gardener/gardener, we maybe have many custom (non-generated) Convert_ funcs and we didn't want to add comment to each one of them. However, in registry-cache we don't have a need for such an exclusion atm.

```suggestion
```